Automattic/wp-calypso
# Add Authorize GitHub card

Closes https://github.com/Automattic/wp-calypso/issues/70376
Related D99604-code

You can now authorise/connect GitHub via the `/hosting-config/:site` page in the same way as via `/marketing/connections/:site`. 

#### Proposed Changes

* Add a GitHub card for authorising WordPress to access GitHub. 
* Conditionally render the card according to the current state. A placeholder view from https://github.com/Automattic/wp-calypso/issues/70378 is rendered if you are already connected. 

Before authorising:
![image](https://user-images.githubusercontent.com/6851384/214771725-1471a9ed-d1ce-4e6e-af95-bbd0d9e220c7.png)

After authorising (just a placeholder):
![image](https://user-images.githubusercontent.com/6851384/214771801-5bcaf4c2-fc06-4a3a-aad2-f0a6f05c0d37.png)


#### Testing Instructions



* Checkout this branch of use Calypso Live
* Follow the test plan at D99604-code, but don't connect via the marketing page
* Go to `/hosting-config/:site` and click "Authorize"
* Authorize WordPress in the popup window
* Verify you see the "Connect Branch" UI after the popup closes. 

#### Pre-merge Checklist





Related to #
